LonaStudio/Workspace/ViewController.swift

```diff
@@ -904,9 +937,18 @@ class ViewController: NSViewController, NSOutlineViewDataSource, NSOutlineViewDe
             if let layer = item as? CSLayer {
                 let checkbox = CheckboxField(frame: NSRect(x: 0, y: 0, width: 20, height: 20))
                 checkbox.value = layer.visible
-                checkbox.onChange = { value in
-                    layer.visible = value
-                    self.render()
+                checkbox.onChange = {[unowned self] value in
+
+                    let oldValue = layer.visible
+                    UndoManager.shared.run(name: "Visible", execute: {[unowned self] in
+                        layer.visible = value
+                        checkbox.state = value ? .on : .off
```

Comment:
We should do `checkbox.value = value` instead of using `.on` and `.off`

Reply:
Unfortunately, It's impossible to assign `Bool` to `NSControl.StateValue` from `checkbox.state` 😅

Reply:
I actually made a subclass to solve this 🤣, just use `value` instead of `state`. I do `checkbox.value = layer.visible` a few lines up

https://github.com/airbnb/Lona/blob/master/LonaStudio/Controls/CheckboxField.swift#L21

Reply:
The only thing I concern, that if I update `checkbox.value`, it triggers the `onChange`, which result in recursive loop.

<img width="1433" alt="screen shot 2018-02-24 at 1 37 48 pm" src="https://user-images.githubusercontent.com/5878421/36626582-ec007c0e-1967-11e8-939d-27b26c5c56e2.png">

Do you have any suggestion?